parlai/crowdsourcing/utils/tests.py

```diff
@@ -107,7 +126,22 @@ def _register_mock_agents(self, num_agents: int = 1) -> List[str]:
 
             # Register the worker
             mock_worker_name = f"MOCK_WORKER_{idx:d}"
-            self.server.register_mock_worker(mock_worker_name)
+            wait_time = 15  # In seconds
+            max_num_tries = 10  # max_num_tries * wait_time is the max time to wait
+            num_tries = 0
+            while num_tries < max_num_tries:
+                try:
+                    self.server.register_mock_worker(mock_worker_name)
+                    break
+                except IndexError:
+                    num_tries += 1
+                    print(
+                        f'A subscriber could not be found after {num_tries:d} '
+                        f'attempt(s). Waiting for {wait_time} seconds...'
+                    )
+                    time.sleep(wait_time)
```

Comment:
Did this end up improving the failed test situation? I'd suggest using a scaling timeout for the wait_time sleep here though as 15 seconds and 10 attempts is a really long time to be waiting. Perhaps something like a 0.5 second wait that doubles 5 or 6 times?

Reply:
Hmm, no, the failed test situation was due to something else (a previous test that wasn't cleaning up the operator properly), but a scaling timeout here is a good idea - changed this